### ISSUE TITLE
Set iteration broken in polyfill (Safari/iOS) for at least babel 5.5.5

### ISSUE THREAD

**Issue by jsg2021** [babel/babel#1712]
I've been experiencing a problem with Set iteration on iOS for a few weeks. Just now got around to identifying my problem.. sorry for the late report. (I'm mostly in Chrome which has native Sets)

```
let a = new Set(['foo', 'bar']);
console.debug('Should see foo, bar:');
for (let v of a) {
	console.log(v);
}
console.debug('did you?');
```

This code exhibits the issue on my local 5.5.5, but the repl on babeljs.io works...i'm assuming that the online repl is just outdated?

**Comment by sebmck** [comment]
How is this an issue with Babel? It's a problem with whatever polyfill you're using.

**Comment by jsg2021** [comment]
I assumed babel provided the ES6 polyfills too

**Comment by sebmck** [comment]
Not unless you're using the polyfill documented [here](http://babeljs.io/docs/usage/polyfill) in which cases it just uses [core-js](https://github.com/zloirock/core-js).

**Comment by jsg2021** [comment]
Yeah, i'm calling on babel's polyfill. So this is an error on core-js?

**Comment by sebmck** [comment]
Looks like it. Babel doesn't have any influence over the runtime.

**Comment by jsg2021** [comment]
Got it, sorry for the noise :)

**Comment by sebmck** [comment]
All good!

**Comment by jsg2021** [comment]
I think my issue was already reported zloirock/core-js#73 :) and looks like it may be fixed soon.

**Comment by zloirock** [comment]
This bug occurs only in very specific cases in the last patch released. It's already fixed in master branch, but I didn't publish it yet because it's not tested in some platforms.

**Comment by zloirock** [comment]
Should be fixed in `0.9.15`.
